### ISSUE TITLE
Feature: keep line number when compiling

### ISSUE THREAD

**Issue by buckle2000** [leafo/moonscript#312]
Since lua is whitespace-insentitive, why not just squeeze code together?

#311

**Comment by buckle2000** [comment]
What we can do furthur, is to trimmer unnecessary spaces.
There is also a bug about multi-line strings, please refer to `BUG` in the code.

**Comment by leafo** [comment]
I think people prefer to have readable code than compressed code. If we were going to compress the code, then we might as well compress it as much as possible.

What's the use case for trying to keep the line numbers? Having stack trace errors that work without rewriting?

**Comment by buckle2000** [comment]
Exactly. Current version cannot rewrite traceback in luajit correctly, in my case, love2d.

**Comment by Guard13007** [comment]
Usually it's not hard to work out the trace from the Lua back to MoonScript.. that said, it would be nice if error messages were more traceable.. but I also like having fairly readable output.

Perhaps a command-line switch to compile in a mode that preserves line numbers? @buckle2000 This might not do what you want though. Sometimes errors in the output Lua aren't so easily traced, and doing things this way would actually make it more difficult to trace because of the loss in readability...

**Comment by buckle2000** [comment]
You are right. It is not hard to find corresponding moonscript from Lua but it's annoying since most bugs are caused by writing mistakes, not logical mistakes, missing comma for example. Given the line number, these bugs can be easily fixed.

**Comment by OttoRobba** [comment]
As an FYI, I have been using buckle's branch and having the correct lines in development so far has been really helpful!

**Comment by buckle2000** [comment]
@OttoRobba Thanks for your proof of concept!

**Comment by sherjilozair** [comment]
@leafo Is it possible to display correct error messages when using moonscript and love2d? Having to read compiled lua code to debug is not fun at all.

**Comment by fendrin** [comment]
@sherjilozair 
Have a look how it is done there:
https://github.com/fendrin/wesnoth-love/blob/master/shared/run_moon.moon

**Comment by ChildishGiant** [comment]
Is there any chance of this ever being merged? It'd make debugging errors far easier as often I find myself with an error on a line that doesn't even exist in the source.

**Comment by leafo** [comment]
`correlate` mode has now been added as a compile option, will be part of moonscript 0.8.0
